src/data/finset/basic.lean

```diff
@@ -1504,6 +1504,221 @@ theorem forall_mem_insert [d : decidable_eq α]
   (∀ x, x ∈ insert a s → p x) ↔ p a ∧ (∀ x, x ∈ s → p x) :=
 by simp only [mem_insert, or_imp_distrib, forall_and_distrib, forall_eq]
 
+lemma eq_or_ne {α : Type*} [D : decidable_eq α] {x y : α} : (x = y)∨ (x ≠ y) :=
```

Comment:
As Mario said, this lemma already exists and `by_cases` is a more convenient way to use it. I still want to use that lemma as an example for remarks about style. A better format for that proof (without changing its steps) would be
```
begin
  have A1 : decidable (x = y) := D x y,
  cases A1,
  { right,
    exact A1, },
  { left,
    exact A1, },
end
```
The changes are:  no `{` or `}` alone on a line; spaces before and after `:` in `A1 : decidable (x = y)`; spaces around `=` in that same expression. Finally, when `apply` closes the goal and can be replaced by `exact`, using `exact` makes the proof more readable.